Expensify/App
# Refactor Report.js to remove Ion.get()

I know this is a huge PR that touches a lot of things, but most of them are pretty superficial. This PR contains:

1. Replaces all references to "report history" with "report actions" (this is the bulk of superficial changes)
1. Removes all `Ion.get()` references in `Report.js`
1. Greatly simplifies all promise chains in `Report.js` to reflect the beauty of Ion being fully asynchronous (no need to do so many things as a sequence of promises)

# Tests
Need to do general regression testing on pretty much everything. Be sure to test the following flow:

1. User A is in the chat app
1. Create a new user B on the same policy as user A
1. In a separate browser, sign into the chat app with User B
1. Have user B use the chat switcher to start a conversation with User A
1. Verify that the chat shows up for User A
1. Verify that notifications work
1. Verify that the unread indicator works

cc @quinthar
